Lib/abc.py

```diff
@@ -121,6 +121,38 @@ def _abc_caches_clear(cls):
             """Clear the caches (for debugging or testing)."""
             _reset_caches(cls)
 
+def update_abstractmethods(cls):
+    """Repopulate the abstract methods of an abstract class, or a subclass of
+    an abstract class.
+
+    If a class has had one of its abstract methods implemented after the
+    class was created, the method will not be considered implemented until
+    this function is called. Alternativly, if a new abstract method has been
+    added to the class, it will only be considered an abstract method of the
+    class after this function is called.
+
+    This function should be called before any use is made of the class,
+    usually in class decorators that add methods to the subject class.
+
+    Returns cls, to allow usage as a class decorator.
+
+    If cls is has any subclasses, raises a RuntimeError.
+    """
+    if cls.__subclasses__():
+        raise RuntimeError("cannot update abstract methods of class after subclassing")
+    abstracts = set()
+    # check the existing abstract methods, keep only the ones that are
+    # still abstract
+    for name in cls.__abstractmethods__:
```

Comment:
Since you didn't check in this function that it's an ABC, it's not guaranteed that cls.__abstractmethods__ works:

>>> class FooABC: pass
... 
>>> FooABC.__abstractmethods__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: __abstractmethods__

Maybe add a test for when this function is called for a non-ABC.